version: '3.6'

services:
  dbeaver:
    image: dbeaver/cloudbeaver:${TAG.CLOUDBEAVER}
    container_name: ${CONTAINER_NAME.CLOUDBEAVER}
    volumes:
      - dbeaver:/opt/cloudbeaver/workspace
    ports:
      - 81:8978
    networks:
      - traefik
      - database_bridge
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.${SERVICE.CLOUDBEAVER}.entrypoints=http"
      - "traefik.http.routers.${SERVICE.CLOUDBEAVER}.rule=Host(`${SERVICE.CLOUDBEAVER}.${FQDN}`)"
      - "traefik.http.middlewares.${SERVICE.CLOUDBEAVER}-https-redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.${SERVICE.CLOUDBEAVER}.middlewares=${SERVICE.CLOUDBEAVER}-https-redirect"
      - "traefik.http.routers.${SERVICE.CLOUDBEAVER}-secure.entrypoints=https"
      - "traefik.http.routers.${SERVICE.CLOUDBEAVER}-secure.rule=Host(`${SERVICE.CLOUDBEAVER}.${FQDN}`)"
      - "traefik.http.routers.${SERVICE.CLOUDBEAVER}-secure.tls=true"
      - "traefik.docker.network=traefik"
      - "flame.type=application"
      - "flame.name=DBeaver"
      - "flame.url=${SERVICE.CLOUDBEAVER}.${FQDN}"
      - "flame.icon=database"
    restart: unless-stopped

volumes:
  dbeaver:
    driver: local
    
networks:
  traefik:
    external: true
  database_bridge:
    external: true
